### DISCUSSION TITLE
Native REST API Authentication 👮‍♂️

### DISCUSSION THREAD

**Discussion by j-vitali** [statamic/cms#9969] [Q&A]
Hi guys, do you plan to add REST API authentication in future?
Or I should build my own?

**Comment by duncanmcclean** [comment]
I don't believe we're planning to add authentication to the REST API in the short term. However, feel free to add your 👍  to the open feature request: statamic/ideas#215.

**Comment by jackmcdade** [comment]
You could also check out https://statamic.com/addons/tv2reg/private-api

**Comment by ghost** [comment]
any updates guys??

**Reply by duncanmcclean** [reply]
Nope. The feature request is still open, you can add a 👍  reaction to it to let us know it's something you want to see.

**Reply by happy-soft-solutions** [reply]
API authentication is not a 'nice to have'; it should be covered by design. For our organization, that should be the only way to use Statamic as a headless CMS in production. Any updates?

**Reply by marcorieser** [reply]
There is the private-api addon, as Jack already [mentioned](https://github.com/statamic/cms/discussions/9969#discussioncomment-9262745).

**Reply by happy-soft-solutions** [reply]
https://statamic.dev/rest-api#authentication: "Coming soon. There are no native access tokens or other common authentication methods ready to use. Yet." Could you prioritize this in the backlog?

**Reply by duncanmcclean** [reply]
Please add a 👍 reaction to the feature request (https://github.com/statamic/ideas/issues/215). We use the reactions to prioritize what goes on our roadmap.

In the meantime, it might be worth looking into [TV2's addon](https://statamic.com/addons/tv2reg/private-api) or implementing it yourself using something like [Laravel Sanctum](https://laravel.com/docs/12.x/sanctum).

**Reply by vitalijalbu** [reply]
Anyone guys have a sanctum ready code to copy paste??

**Reply by remluben** [reply]
Any news on this? 😄

**Reply by vitalijalbu** [reply]
> Any news on this? 😄

i did it using sanctum, pretty simple...

**Reply by duncanmcclean** [reply]
Sorry, there hasn't been any movement on this. 

Leave a 👍 reaction and "Subscribe" to the [the feature request](https://github.com/statamic/ideas/issues/215) to be notified when there's any progress.

**Reply by remluben** [reply]
> > Any news on this? 😄
> 
> i did it using sanctum, pretty simple...

Did you also turn of Statamic caching for `api` routes to fully protect? I ran into issues where after an authenticated and authorized user loaded resources from an api endpoint, Statamic made cached response take priority over auth middleware, thus returned data even for unauthenticated users. 

It would be nice to know if anyone else needed this in _config/statamic/api.php_
```
    'cache' => [
        'expiry' => false, // Disable caching to prevent serving cached content to unauthenticated users
    ],
```

**Reply by ryanmitchell** [reply]
Yes you need to turn off caching
